R/viz.R

```diff
@@ -2742,16 +2746,34 @@ splendidHeatmap = function(this_matrix,
 #' sv_plot = fancy_sv_chrdistplot(this_sample = "HTMCP-01-06-00422-01A-01D")
 #'
 fancy_sv_chrdistplot = function(this_sample,
+                                maf_data,
+                                maf_path = NULL,
+                                variant_type_col = 10,
+                                chromosome_col = 5,
                                 plot_title = paste0(this_sample),
                                 plot_subtitle = "Structural Variant Distribution Per Chromosome",
                                 chr_select = paste0("chr", c(1:22)),
                                 coding_only = FALSE,
                                 from_flatfile = TRUE,
                                 use_augmented_maf = TRUE){
 
+  if(!missing(maf_data)){
+    maf = maf_data
+    maf = as.data.frame(maf)
+    colnames(maf)[variant_type_col] = "Variant_Type"
+    colnames(maf)[chromosome_col] = "Chromosome"
+
+  }else if(!is.null(maf_path)){
+    maf = fread_maf(maf_path)
+    maf = as.data.frame(maf)
+    colnames(maf)[variant_type_col] = "Variant_Type"
+    colnames(maf)[chromosome_col] = "Chromosome"
+  }
 
   #get maf data for a specific sample.
-  maf = assign_cn_to_ssm(this_sample = this_sample, coding_only = coding_only, from_flatfile = from_flatfile, use_augmented_maf = use_augmented_maf)$maf
+  if(missing(maf_data) && is.null(maf_path)){
+    maf = assign_cn_to_ssm(this_sample = this_sample, coding_only = coding_only, from_flatfile = from_flatfile, use_augmented_maf = use_augmented_maf)$maf
```

Comment:
What happens here if maf_path is provided? It looks like the code isn't handling that case.

Reply:
I think it does, after some testing everything seems to be handled the right way:

```
#get maf as df in R.
maf = assign_cn_to_ssm(this_sample = this_sample, coding_only = FALSE, from_flatfile = TRUE, use_augmented_maf = TRUE)$maf

#export maf (to read back into r with maf_path).
write.table(maf, "../../maf.maf", quote = FALSE, sep = "\t", row.names = FALSE, col.names = TRUE)

#using maf_path to build plot.
sv_plot_maf_path = fancy_sv_chrdistplot(this_sample = "", plot_subtitle = "maf_path to build plot", maf_path = "../../maf.maf")

#using maf_data to build plot.
sv_plot_maf_data = fancy_sv_chrdistplot(this_sample = "", plot_subtitle = "maf_data to build plot", maf_data = maf)

#using assign_cn_to_ssm to retrieve plotting data.
sv_plot_asssign_cn_to_ssm = fancy_sv_chrdistplot(this_sample = "HTMCP-01-06-00422-01A-01D", plot_title = "", plot_subtitle = "assign_cn_to_ssm to retrieve plotting data")
```
![Screenshot 2022-05-10 at 12 26 40](https://user-images.githubusercontent.com/32147626/167707062-11abf5de-e14f-41fd-bc97-52fc5c703c1b.png)

Reply:
Grat